libs/agno/agno/agent/agent.py

```diff
@@ -4255,66 +4444,183 @@ def get_tool_call_history(num_calls: int = 3) -> str:
 
         return get_tool_call_history
 
-    def search_knowledge_base(self, query: str) -> str:
-        """Use this function to search the knowledge base for information about a query.
+    def search_knowledge_base_function(
+        self, knowledge_filters: Optional[Dict[str, Any]] = None, async_mode: bool = False
+    ) -> Callable:
+        """Factory function to create a search_knowledge_base function with filters."""
+        # Determine which filters to use
+        effective_filters = knowledge_filters if knowledge_filters is not None else self.knowledge_filters
```

Comment:
Wasn't this determined earlier?

Reply:
no but this is for if we use the `run` filters or `Agent` level filters, am i repeating it anywhere?

Reply:
actually good catch, resolved